helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataReader.java

```diff
@@ -85,23 +85,14 @@ public ZkRoutingDataReader(String namespace, String zkAddress,
           "Routing data directory ZNode " + MetadataStoreRoutingConstants.ROUTING_DATA_PATH
               + " does not exist. Routing ZooKeeper address: " + _zkAddress);
     }
-    if (children == null || children.isEmpty()) {
-      throw new InvalidRoutingDataException(
-          "There are no metadata store realms defined. Routing ZooKeeper address: " + _zkAddress);
-    }
     for (String child : children) {
```

Comment:
Nit: let's name our variables better. 

for (String realmName : allRealmNames) {

might be a better pick?

Reply:
Agreed.